app/policies/application_policy.rb

```diff
@@ -40,6 +40,10 @@ def destroy?
   private
 
   def admin_or_current_user?
-    user.admin? or record.id == user.id or record.user.id == user.id
+    if record.class == Invoice
+      user.admin? or record.user.id == user.id
+    else
+      user.admin? or record.id == user.id
+    end
```

Comment:
```suggestion
    user.admin? or record.id == user.id or (record.respond_to(:user) && record.user.id == user.id)
```

Reply:
Well thought! That makes more sense when thinking about "hard code"!